hydra-access-controls/app/models/hydra/access_controls/permission.rb

```diff
@@ -1,7 +1,19 @@
 module Hydra::AccessControls
+  AGENT_URL_SCHEME = 'http'.freeze
+  AGENT_URL_HOST = 'projecthydra.org'.freeze
+  AGENT_URL_PATH = '/ns/auth/'.freeze
+  GROUP_AGENT_PATH = '/ns/auth/group'.freeze
+  PERSON_AGENT_PATH = '/ns/auth/person'.freeze
   AGENT_URL_PREFIX = 'http://projecthydra.org/ns/auth/'.freeze
-  GROUP_AGENT_URL_PREFIX = 'http://projecthydra.org/ns/auth/group'.freeze
-  PERSON_AGENT_URL_PREFIX = 'http://projecthydra.org/ns/auth/person'.freeze
+
+  GroupAgentUrlPrefix = Exception.new("GROUP_AGENT_URL_PREFIX is deprecated and being replaced by the individual portions of the url, AGENT_URL_SCHEME, AGENT_URL_HOST, and GROUP_AGENT_PATH")
+  GROUP_AGENT_URL_PREFIX = GroupAgentUrlPrefix # previous version used this name, superceded by individual url components
+  PersonAgentUrlPrefix = Exception.new("PERSON_AGENT_URL_PREFIX is deprecated and being replaced by the individual portions of the url, AGENT_URL_SCHEME, AGENT_URL_HOST, and PERSON_AGENT_PATH")
+  PERSON_AGENT_URL_PREFIX = PersonAgentUrlPrefix
+
+  deprecate_constant :GROUP_AGENT_URL_PREFIX
+  deprecate_constant :PERSON_AGENT_URL_PREFIX
```

Comment:
I like the deprecation warning.  I've never done this before.  If someone has overridden `GROUP_AGENT_URL_PREFIX` will this present the warning message but still work or does it just stop working and give the warning message?

Reply:
So, it looks like adding these deprecations has broken a bunch of tests, and just makes use of those constants stop working while giving the warning, which wasn't really my intention. I'm reverting that particular commit.